kpi/management/commands/sync_form_media.py

```diff
@@ -0,0 +1,194 @@
+                ).exists()
+            ):
+                continue
+
+            media_content = _get_media_file_content(
+                token=token,
+                username=user.username,
+                asset_uid=asset.uid,
+                media_id=media_id,
+            )
+
+            af = AssetFile.objects.create(
+                asset=asset,
+                user=user,
+                content=ContentFile(media_content, name=media_filename),
+                file_type=AssetFile.FORM_MEDIA,
+                description='default',
+            )
+            synced_files.append(
+                {
+                    'data_value': media_filename,
+                    'id': media_id,
+                }
+            )
+
+        kpi_after = AssetFile.objects.filter(
+            asset=asset,
+            file_type=AssetFile.FORM_MEDIA,
+            date_deleted__isnull=True,
+        ).count()
+        sync_stats = {
+            'asset_uid': asset.uid,
+            'xformid': asset_xform_id,
+            'asset_owner__username': asset.owner.username,
+            'kobocat_form_media': len(
+                [m for m in media_files if m['xform'] == asset_xform_id]
+            ),
+            'kpi_form_media_pre_sync': kpi_before,
+            'kpi_form_media_post_sync': kpi_after,
+            'kpi_form_media_difference': kpi_after - kpi_before,
+        }
+        if verbosity == 3:
+            sync_stats['synced_files'] = synced_files
+        if verbosity > 1:
+            sync_stats_all.append(sync_stats)
+
+    stats_out = {
+        'assets_all_count': assets_all_count,
+        'assets_selected_count': assets_selected_count,
+    }
+    if verbosity > 1:
+        stats_out['sync_stats'] = sync_stats_all,
+
+    return stats_out
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--asset-uid',
+            action='store',
+            dest='asset_uid',
+            default=None,
+            help="Sync only a specific asset's form-media"
+        )
+        parser.add_argument(
+            '--username',
+            action='store',
+            dest='username',
+            default=None,
+            help="Sync only a specific user's form-media for assets that they own"
+        )
+        parser.add_argument(
+            '--quiet',
+            action='store_true',
+            dest='quiet',
+            default=False,
+            help='Do not output status messages'
+        )
+
+    def handle(self, *args, **options):
+        if not settings.KOBOCAT_URL or not settings.KOBOCAT_INTERNAL_URL:
+            raise ImproperlyConfigured(
+                'Both KOBOCAT_URL and KOBOCAT_INTERNAL_URL must be '
+                'configured before using this command'
+            )
+        quiet = options.get('quiet')
+        verbosity = options.get('verbosity')
+
+        stats = _sync_media_files(**options)
+
+        if not quiet:
```

Comment:
_Small optimization:_ 
All stats queries are run on the DB even if `quiet` is True. I think it would be better to pass `quiet` to `_sync_media_files` too and skip not-required queries when `quiet` is True.